lib/model/message_list.dart

```diff
@@ -222,6 +222,17 @@ mixin _MessageSequence {
     return binarySearchByKey(items, messageId, _compareItemToMessageId);
   }
 
+  List<Message>? getMessagesSublist(int firstMessageId, int lastMessageId) {
+    assert(firstMessageId <= lastMessageId);
+    final firstIndex = _findMessageWithId(firstMessageId);
+    final lastIndex = _findMessageWithId(lastMessageId);
+    if (firstIndex == -1 || lastIndex == -1) {
+      // TODO(log)
+      return null;
+    }
+    return messages.sublist(firstIndex, lastIndex + 1);
```

Comment:
This is making a new List by copying the given range of elements. But it looks like the one caller immediately consumes it as effectively an Iterable. So this could avoid the copy by instead using `List.getRange`.